Exception in translate() on collection of coding followed by first()

Evaluation of:

```
code.coding.translate('http://aehrc.com/fhir/ConceptMap/aehrc-snomap-starter', false, 'wider').first().display
```

fails with:
```
Caused by: java.lang.IllegalStateException: Couldn't find Condition#184 in [_extract_id#206]
	at org.apache.spark.sql.catalyst.expressions.BindReferences$$anonfun$bindReference$1.applyOrElse(BoundAttribute.scala:80)
	at org.apache.spark.sql.catalyst.expressions.BindReferences$$anonfun$bindReference$1.applyOrElse(BoundAttribute.scala:73)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDownWithPruning$1(TreeNode.scala:461)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(origin.scala:76)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDownWithPruning(TreeNode.scala:461)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDownWithPruning$3(TreeNode.scala:466)
	at org.apache.spark.sql.catalyst.trees.UnaryLike.mapChildren(TreeNode.scala:1215)
	at org.apache.spark.sql.catalyst.trees.UnaryLike.mapChildren$(TreeNode.scala:1214)
```

**To Reproduce**

Run: https://gist.github.com/piotrszul/900b0279947d9649b36661a417c61755
on python example data.

Present at revision: 0058507b8a (on release/8.0.0)

**Expected behavior**
Should produce a valid result, same as:
```
code.coding.translate('http://aehrc.com/fhir/ConceptMap/aehrc-snomap-starter', false, 'wider').display.first()
```
which works.

--- a/fhirpath/src/test/resources/viewTests/misc.json
+++ b/fhirpath/src/test/resources/viewTests/misc.json
@@ -62,6 +62,36 @@
           "empty": null
         }
       ]
+    },
+    {
+      "title": "traversal on array first()",
+      "view": {
+        "resource": "Patient",
+        "select": [
+          {
+            "column": [
+              {
+                "path": "id",
+                "name": "id"
+              },
+              {
+                "path": "name.first().family",
+                "name": "given_name"
+              }
+            ]
+          }
+        ]
+      },
+      "expect": [
+        {
+          "id": "p1",
+          "given_name": "f1"
+        },
+        {
+          "id": "p2",
+          "given_name": "f2"
+        }
+      ]
     }
   ]
 }

--- a/lib/python/requirements/package.txt
+++ b/lib/python/requirements/package.txt
@@ -1,2 +1,2 @@
-pyspark==3.5.1
+pyspark==3.5.5
 deprecated==1.2.14

--- a/lib/python/setup.py
+++ b/lib/python/setup.py
@@ -62,7 +62,7 @@ setup(
     ],
     license="Apache License, version 2.0",
     python_requires=">=3.8",
-    install_requires=["pyspark>=3.5.0,<3.6.0", "deprecated>=1.2.13"],
+    install_requires=["pyspark>=3.5.2,<3.6.0", "deprecated>=1.2.13"],
     include_package_data=True,
     data_files=[
         ("share/pathling/examples", glob.glob("examples/*.py")),

--- a/lib/python/tests/test_view.py
+++ b/lib/python/tests/test_view.py
@@ -1,0 +1,47 @@
+#  Copyright 2023 Commonwealth Scientific and Industrial Research
+#  Organisation (CSIRO) ABN 41 687 119 230.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import os
+from unittest.mock import Mock, patch
+from pyspark.sql import Row, DataFrame
+from pytest import fixture
+
+
+@fixture(scope="module")
+def ndjson_test_data_dir(test_data_dir):
+    return os.path.join(test_data_dir, "ndjson")
+
+
+ResultRow = Row("id", "family_name")
+
+
+def test_view_on_ndjson(ndjson_test_data_dir, pathling_ctx):
+    data_source = pathling_ctx.read.ndjson(ndjson_test_data_dir)
+    result = data_source.view(
+        resource='Patient',
+        select=[
+            {
+                'column': [
+                    {'path': 'id', 'name': 'id'},
+                    {'path': "name.first().family", 'name': 'family_name'}
+                ]
+            }
+        ]
+    )
+    assert result.columns == list(ResultRow)
+    assert result.limit(2).collect() == [
+        ResultRow('8ee183e2-b3c0-4151-be94-b945d6aa8c6d', 'Krajcik437'),
+        ResultRow('beff242e-580b-47c0-9844-c1a68c36c5bf', 'Towne435'),
+    ]

